picosat.c

```diff
@@ -2382,7 +2382,7 @@ add_simplified_clause (PS * ps, int learned)
 	  for (p = ps->added; p < ps->ahead; p++) 
 	    {
 	      lit = *p;
-	      if (lit->val)
+	      if (lit->val && ps->levels)
```

Comment:
This one is concerning; it appears a few lines later that `ps->levels` is unconditionally dereferenced. TBH, I don't understand how this results in merely a ubsan runtime warning and not a SEGV. Unfortunately, my knowledge of this code is essentially non-existent.

Reply:
@soonhokong I see you merged the patch. Did you see the above discussion?